### ISSUE TITLE
Feature idea - pre & post command hooks

### ISSUE THREAD

**Issue by sanchothefat** [wp-cli/wp-cli#2731]
Admittedly there's only a single use-case I can think of for this right now but others might be able to think of more so this is to gauge interest really.

In an AWS auto-scaling EC2 cluster you may start a long running CLI command, there's no guarantee that the instance won't be cycled while the command is running unless you run an AWS CLI command to prevent that.

It'd be cool to define commands in the yaml config file that are run before and or after any WP CLI commands.

**Comment by danielbachhuber** [comment]
> It'd be cool to define commands in the yaml config file that are run before and or after any WP CLI commands.

You can do this in v0.24.0-alpha #2647. Here's an example:

```
WP_CLI::add_hook( 'after_invoke:site empty', function(){
	global $wpdb;
	foreach( array( 'p2p', 'p2pmeta' ) as $table ) {
		$table = $wpdb->$table;
		$wpdb->query( "TRUNCATE $table" );
	}
});
```

If you don't want to commit that to your project, you can include it at runtime by putting it in a file, and running `wp --require=file.php site empty`

**Comment by sanchothefat** [comment]
ah cool, wasn't aware of this, thanks!

On 10 May 2016 at 19:53, Daniel Bachhuber <notifications@github.com> wrote:

> Closed #2731 <https://github.com/wp-cli/wp-cli/issues/2731>.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/wp-cli/wp-cli/issues/2731#event-656510881>
>



-- 
Robert O'Rourke
*mobile:* +447816329424
*skype: *rob.o.rourke
*web:* röb.co
